src/taskpane/pages/dashboard.tsx

```diff
@@ -1,46 +1,53 @@
-import {
-  IStackStyles,
-  DefaultPalette,
-  ISearchBoxStyles,
-  Stack,
-  SearchBox,
-  PrimaryButton,
-  ButtonType,
-} from "@fluentui/react";
-import React from "react";
-import Header from "../components/Header";
-/* global Word */
+import React, { useState } from "react";
+import data from "../../utils/data";
+import ReferenceList from "../components/ReferenceList";
+import SearchField from "../components/SearchField";
+// /* global Word */
 
-const stackStyles: IStackStyles = {
-  root: {
-    background: DefaultPalette.white,
-  },
-};
-const searchBoxStyles: Partial<ISearchBoxStyles> = { root: {} };
+function containsSearchTerm(keyword: string) {
+  return function (item) {
+    try {
+      return (
+        item.title.toLowerCase().indexOf(keyword.toLowerCase()) >= 0 ||
+        item.author.toLowerCase().indexOf(keyword.toLowerCase()) >= 0 ||
+        item.year.indexOf(keyword) >= 0
+      );
+    } catch (e) {
+      return false;
+    }
```

Comment:
Is the try-catch statement to prevent throwing an error if the title/author/year doesn't exist?

Would `item.title?.toLowerCase().includes(keyword.toLowerCase())` work?

Reply:
Is it worth the effort of doing the Arrays.some approach if we intend to modify the list more? 

https://stackoverflow.com/questions/62484862/optional-chaining-in-combination-with-tolowercase-and-indexof-during-array-map